pkgs/development/python-modules/libmambapy/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib
+, fetchFromGitHub
+, buildPythonPackage
+, cmake
+, ninja
+, libmamba
+, python
+, pybind11
+, setuptools
+, wheel
+, scikit-build
+, breakpointHook
+, ...
+}:
+buildPythonPackage rec {
+  pname = "libmambapy";
+  version = "1.5.7";
+  pyproject = true;
+  nativeBuildInputs = [ cmake ninja breakpointHook ];
+  buildInputs = [ python libmamba pybind11 ];
+  src = fetchFromGitHub {
+    owner = "mamba-org";
+    repo = "mamba";
+    rev = "${pname}-${version}";
+    hash = "sha256-HfmvLi9IBWlaGAn2Ej4Bnm4b3l19jEXwNl5IUkdVxi0=";
+  };
+
+  souceRoot = "${src.name}/libmambapy";
+
+  preBuild = ''
+    mkdir cmake-build
+    cd cmake-build
+    cmake $src -GNinja -DBUILD_LIBMAMBAPY=on
```

Comment:
You could still pass `$cmakeFlags` here and define then as an attribute pass to `mkDerivation`. Idk how to do this well, but here's possibly a similar case: https://github.com/SomeoneSerge/dust3r.nix/blob/4fd7d61310f2289ba9843c9854aaaf59837eafbd/nix/magnum-bindings/package.nix#L41-L50

Also I noticed `libmambapy` uses `skbuild`, which means you can pass most of the flags by re-exporting `cmakeFlags` as `CMAKE_ARGS`: https://github.com/NixOS/nixpkgs/blob/76de45b55faf23da7cad2a2ca1136da5cd4df95b/pkgs/development/python-modules/mitsuba3/default.nix#L90-L104

("most" rather than "all" because space-splitting and `cmakeFlagsArray`...)

Reply:
`skbuild` doesn't get used. I didn't know about the `removePythonDeps`.